File: src/commands/publish.ts
import { queriesInit } from '@/commands/use'

export type EventPublishValue = {
  name: string;
  value: string;
}
export type EventPublishRequest = {
  name: string;
  value: EventPublishValue[],
}
export const {
  useEventPublishQuery,
  useEventPublishLazy,
} = queriesInit<EventPublishRequest, {}>('event_publish')

File: src-tauri/command/publish.rs
use serde::{Serialize, Deserialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct EventPublishValue {
    name: String,
    value: String,
}
#[derive(Debug, Serialize, Deserialize)]
pub struct EventPublishRequest {
    name: String,
    value: Vec<EventPublishValue>, // like Note { name, dscription, project, save path }
}
#[tauri::command]
pub fn event_publish(event: EventPublishRequest) {
    println!("{:?}", event);
}
